### ISSUE TITLE
Some website fonts are too faded

### ISSUE THREAD

**Issue by Jolly-Pirate** [MrAlex94/Waterfox-Old#153]
**Describe the bug**
Some website fonts are too faded, they look clearer in other browsers. I'm using Waterfox Classic 2022.08.

**To Reproduce**
Examples:
https://www.apache.org/dyn/closer.cgi/netbeans/netbeans/15/netbeans-15-bin.zip
https://www.reasonablefaith.org/media/reasonable-faith-podcast/sir-karl-popper-an-interesting-philosopher

**Screenshots**

- In Waterfox Classic
![image](https://user-images.githubusercontent.com/4613678/188943132-c2516860-a1b6-4a81-9bd3-c1fe06c05fb8.png)

- In Chrome and Edge
![image](https://user-images.githubusercontent.com/4613678/188943252-1fc4a120-d014-4012-a215-4c9b4d94561d.png)

**Desktop**
 - OS: Windows 10
 - Version 21H2

**Comment by linkoviccsumoszky** [comment]
Same here. For a workaround, try some user agent switcher and set it to an earlier version of Firefox (60.8 works for me now).

**Comment by lokpro** [comment]
![image](https://user-images.githubusercontent.com/2010297/189067997-aa5ebc42-c3d4-429f-923f-d66aa24a4d5b.png)

**Comment by Jolly-Pirate** [comment]
> Same here. For a workaround, try some user agent switcher and set it to an earlier version of Firefox (60.8 works for me now).

That worked, for now, as well as switching the agent to google bot or curl. However, other agents/versions didn't work.

**Comment by Felipefpl** [comment]
Tested and i couldnt reproduce the problem here, did you change any of the font's settings in the browser or some extension did it for you?

**Comment by Jolly-Pirate** [comment]
I have several extensions but they have nothing to do with fonts. I even created a default profile for testing, with no extension at all, and the problem persists.

**Comment by Squall-Leonhart** [comment]
These fonts aren't faded, the glyph is corrupted due to display driver forced settings.

stop using driver forced FXAA.

Otherwise, it looks like semibold fonts are missing.

**Comment by Jolly-Pirate** [comment]
@Squall-Leonhart FXAA is already OFF in my Nvidia settings, and having hardware acceleration on or off in the browser doesn't change a thing. Besides, fonts are system-wide, so how can one browser have trouble rendering a font when the other browsers are fine?

I've further tested with Waterfox current, and it has no problem rendering the fonts. In fact, it renders this page https://www.reasonablefaith.org/media/reasonable-faith-podcast/sir-karl-popper-an-interesting-philosopher far better than Chrome, Vivaldi, Waterfox Classic.

**Comment by Squall-Leonhart** [comment]
nonsense, there are no issues with the site on classic across any supported OS, issue is with your systems and not a waterfox bug.

**Comment by Squall-Leonhart** [comment]
Montserrat is broken on windows 10, the site is to blame.

**Comment by linkoviccsumoszky** [comment]
> nonsense, there are no issues with the site on classic across any supported OS, issue is with your systems and not a waterfox bug.

not true. lots of other sites had the same issue - user agent switcher fixed the problem as said above. it has nothing to do with graphics settings.

**Comment by FeepingCreature** [comment]
I have the same issue. Mesa graphics (AMD), so not an Nvidia/FXAA thing. Can confirm that user-agent Firefox 60.8 "fixes" it. I've seen it happen with Montserrat and Josefin Sans.

**Comment by buttercookie42** [comment]
The issue is that Waterfox Classic doesn't support variable fonts at the moment (i.e. one single font file with the basic glyph shape plus instructions on how to scale that glyph shape depending on the font weight, instead of a separate font file for each font weight).

Google Fonts apparently does some UA sniffing for Firefox (and possibly other major browsers), so only Firefox versions that support rendering variable fonts actually get served variable fonts, but for Waterfox they don't bother and serve variable fonts regardless.

The failure mode is presumably the raw base shape of each glyph is being rendered, which apparently corresponds to the thinnest font weight supported by that respective font.
